services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_service_specs/volume_remover.py

```diff
@@ -0,0 +1,147 @@
+    """
+    Extracts `XX.XX.XX` where X is a range [0-9] from
+    a given docker version
+    """
+
+    @classmethod
+    def __get_validators__(cls):
+        yield cls.validate_docker_version
+
+    @classmethod
+    def validate_docker_version(cls, docker_version: str) -> str:
+        try:
+            search_result = re.search(r"^\d\d.\d\d.\d\d", docker_version)
+            return search_result.group()
+        except AttributeError:
+            raise ValueError(  # pylint: disable=raise-missing-from
+                f"{docker_version} appears not to be a valid docker version"
+            )
+
+
+# NOTE: below `retry` function is inspired by
+# https://gist.github.com/sj26/88e1c6584397bb7c13bd11108a579746
+SH_SCRIPT_REMOVE_VOLUMES = """
+set -e;
+
+error_counter=0
+
+function retry {{
+  local retries={retries}
+  shift
+
+  local count=0
+  until "$@"; do
+    exit=$?
+
+    count=$(($count + 1))
+    if [ $count -lt $retries ]; then
+      echo "Retry $count/$retries exited $exit, retrying in {sleep} seconds..."
+      sleep {sleep}
+    else
+      echo "Retry $count/$retries exited $exit, no more retries left."
+      let error_counter=error_counter+1
+      return 0
+    fi
+  done
+  return 0
+}}
+
+for volume_name in {volume_names_seq}
+do
+    retry 3 docker volume rm "$volume_name"
+done
+
+if [ "$error_counter" -ne "0" ]; then
+    echo "ERROR: Please check above logs, there was/were $error_counter error/s."
+    exit 1
+fi
+"""
+
+
+def spec_volume_removal_service(
+    dynamic_sidecar_settings: DynamicSidecarSettings,
+    docker_node_id: str,
+    volume_names: list[str],
+    docker_version: DockerVersion,
+    *,
+    volume_removal_attempts: int,
+    sleep_between_attempts_s: int,
+    service_timeout_s: int,
+) -> AioDockerServiceSpec:
+    """
+    Generates a service spec for with base image
+    `docker:{docker_version}-dind` running the above bash script.
+
+    The bash script will attempt to remove each individual volume
+    a few times before giving up.
+    The script will exit with error if it is not capable of
+    removing the volume.
+
+    NOTE: expect the container of the service to exit with code 0,
+    otherwise there was an error.
+    NOTE: the bash script will exit 1 if it cannot find a
+    volume to remove.
+    NOTE: service must be removed once it finishes or it will
+    remain in the system.
+    """
+
+    volume_names_seq = " ".join(volume_names)
+    formatted_command = SH_SCRIPT_REMOVE_VOLUMES.format(
+        volume_names_seq=volume_names_seq,
+        retries=volume_removal_attempts,
+        sleep=sleep_between_attempts_s,
+    )
+    logger.debug("Service will run:\n%s", formatted_command)
+    command = ["sh", "-c", formatted_command]
+
+    create_service_params = {
+        "labels": {
+            "volume_names": json.dumps(volume_names),
+            "volume_removal_attempts": f"{volume_removal_attempts}",
+            "sleep_between_attempts_s": f"{sleep_between_attempts_s}",
```

Comment:
you might also add the usual labels just in case we need to trace it.

Reply:
added this